terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-up.ts

```diff
@@ -248,20 +291,41 @@ async function allRunnersBusy(
     return 0;
   }
 
-  // Have a fail safe just in case we're likely to need more runners
+  if (requestedCount > maxAllowedScaleUp) {
+    console.info(
+      `Requested count ${requestedCount} is higher than max allowed scale up ${maxAllowedScaleUp}, ` +
+        `will scale up ${maxAllowedScaleUp} instead`,
+    );
+    requestedCount = maxAllowedScaleUp;
+  }
+
   const availableCount = runnersWithLabel.length - busyCount;
-  // Min runners for scale-up must be at least 1 otherwise scale-up won't ever increase runners
-  const minRunners = Config.Instance.minAvailableRunners > 0 ? Config.Instance.minAvailableRunners : 1;
-  if (availableCount < minRunners) {
-    console.info(`Available (${availableCount}) runners is below minimum ${minRunners}`);
-    // It is impossible to accumulate runners if we know that the one we're creating will be terminated.
+  let additionalNeeded = requestedCount - availableCount;
+
+  if (additionalNeeded > 0) {
+    // We need to scale up to process the request
+    // TODO: See if we should scale up extra runners for ephemerals
     if (isEphemeral) {
-      const ratio: number = availableCount / (minRunners * 1.5);
-      return Math.random() < ratio ? 3 : 1;
-    } else {
-      return 1;
+      additionalNeeded = getOverprovisionedCountForEphemeralRunner(additionalNeeded);
     }
+
+    return additionalNeeded;
```

Comment:
IIUC this will scale always to only support current need, never maintaining some stand by buffer of runners.

Reply:
Hmmm...you're right. I need to rework this logic a bit.